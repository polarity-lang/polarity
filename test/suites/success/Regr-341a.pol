data Top { Unit }

codata Fun(a b: Type) {
    Fun(a,b).ap(a b: Type, x: a) : b
}

codef F: Fun(Type, Type) {
    .ap(_,_,x) => Type
}

let foo(t: Type, x: Fun(t, Type)): Type {
    ?
}

let T : Type {
    foo(F.ap(Type, Type, (? : Type)), \x. F.ap(Type, Type, x))
}
