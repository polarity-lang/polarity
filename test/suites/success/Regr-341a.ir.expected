codef F { .ap(x, x0, x1) => <ZST> }

let foo(t, x) { panic!("not yet implemented") }

let T {
    foo(F.ap(<ZST>, <ZST>, panic!("not yet implemented")),
        comatch { .ap(_, _, x1) => F.ap(<ZST>, <ZST>, x1) })
}
